python/tank_vendor/shotgun_base/utils.py

```diff
@@ -18,6 +18,34 @@
 from .log import get_shotgun_base_logger
 log = get_shotgun_base_logger()
 
+
+def with_cleared_umask(func):
+    """
+    Decorator which clears the umask for a method.
+
+    The umask controls default permissions for files
```

Comment:
is that really necessary? This is not a normal function, never meant to be called as a function!